packages/app/src/systems/Transaction/components/TxRecipientCard/TxRecipientCard.tsx

```diff
@@ -29,8 +40,41 @@ export const TxRecipientCard: TxRecipientCardComponent = ({
   const isContract = recipient?.type === AddressType.contract;
   const isEthChain = recipient?.chain === ChainName.ethereum;
   const isNetwork = address === 'Network';
-  const name =
-    accounts?.find((a) => a.address === fuelAddress)?.name || 'unknown';
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+  useEffect(() => {
+    const setAccountName = () => {
+      const accountName =
+        accounts?.find((a) => a.address === fuelAddress)?.name || 'unknown';
+      setName(accountName);
+    };
+
+    const resolveAddressName = async () => {
+      const handleName = await reverseResolver(fuelAddress);
+
+      if (handleName) {
+        return setName(`@${handleName}`);
+      }
+
+      setAccountName();
+    };
+
+    if (!name && isReceiver) {
+      resolveAddressName();
+    }
+
+    if (!isReceiver) {
+      setAccountName();
+    }
```

Comment:
I added it only to show in the receiver, thinking about the transaction sending flow, what do you think?

If they're not a receiver, there could be a conflict with the wallet name.